### PR TITLE
Add tests around hashcode collisions

### PR DESCRIPTION
**Pull request by MansurAshraf**
potential bug in ProductLikeMacro's hashcode method

### CODE REVIEW

File: scalding-serialization-macros/src/test/scala/com/twitter/scalding/serialization/macros/MacroOrderingProperties.scala [hunk 1]
```diff
@@ -238,40 +248,49 @@ class MacroOrderingProperties extends FunSuite with PropertyChecks with ShouldMa
   test("Test out jl.Byte") {
     implicit val a = arbMap { b: Byte => java.lang.Byte.valueOf(b) }
     check[java.lang.Byte]
+    checkCollisions[java.lang.Byte]
```
Comment (johnynek) [inline scalding-serialization-macros/src/test/scala/com/twitter/scalding/serialization/macros/MacroOrderingProperties.scala]: there are only 256 bytes, so taking 100 of them, the chance of collision seems pretty high. Seems like we should use the formula, pass the cardinality of T into checkCollisions, and then have it pick the k (100 in your example).

Comment (johnynek) [inline scalding-serialization-macros/src/test/scala/com/twitter/scalding/serialization/macros/MacroOrderingProperties.scala]: actually, if we are hashing 256 things into 32 bits, it should probably not collide actually.

So, this test is fine (assuming it passes).

**Comment by johnynek** [comment]
wait, this test passed?

**Comment by johnynek** [comment]
dammit, I bet we are not testing scalding-serialization-macros.

**Comment by MansurAshraf** [comment]
arrgh, resubmitting

**Pull request merged**